app/controllers/close_contacts_controller.rb

```diff
@@ -42,4 +42,22 @@ def update
                                created_by: current_user.email,
                                comment: "User edited a close contact (ID: #{cc.id}).")
   end
+
+  # Delete an existing close contact record
+  def destroy
+    redirect_to root_url && return unless current_user.can_create_patient_close_contacts?
```

Comment:
can you add all these checks as `before_actions` (user can create, valid patient and valid cc)?  See @mayerm94's PR for how do do this